src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1253,17 +1253,41 @@ public static bool IsPossibleTupleContext(this SyntaxTree syntaxTree, SyntaxToke
             return false;
         }
 
-        public static bool IsPatternContext(this SyntaxTree syntaxTree, SyntaxToken leftToken, int position)
+        public static bool IsAtStartOfPattern(this SyntaxTree syntaxTree, SyntaxToken leftToken, int position)
         {
             leftToken = leftToken.GetPreviousTokenIfTouchingWord(position);
 
+            // If we're dealing with an expression surrounded by one or more sets of open parentheses, we need to
+            // walk up the parens in order to see if we're actually at the start of a valid pattern or not.
+            // After recursing, we should be left with at most a single open paren to analyze.
+            if (leftToken.IsKind(SyntaxKind.OpenParenToken))
+            {
+                if (leftToken.Parent.IsParentKind(SyntaxKind.ParenthesizedExpression))
+                {
+                    return IsAtStartOfPattern(syntaxTree, leftToken.Parent.Parent.GetFirstToken(), leftToken.Parent.Parent.SpanStart);
+                }
+
+#if !CODE_STYLE
+                if (leftToken.Parent.IsParentKind(SyntaxKind.ParenthesizedPattern))
+                {
+                    return IsAtStartOfPattern(syntaxTree, leftToken.Parent.Parent.GetFirstToken(), leftToken.Parent.Parent.SpanStart);
+                }
```

Comment:
if the left token is a parenthesized pattern you can just immediatley return 'true' here, you know you're at the start of a pattern right?  no need to recurse.